samples/vase-component/src/vase_component/main.clj

```diff
@@ -0,0 +1,26 @@
+(ns vase-component.main
+  "Main entry-point for the service when deployed."
+  (:gen-class)
+  (:require
+   [com.stuartsierra.component :as component]
+   [vase-component.system :as system]
+   [vase-component.endpoint :as endpoint]
+   [io.pedestal.log :as log]))
+
+(defn- env-port
+  "Gets port from environment variable, or throws."
+  []
+  (if-let [s (System/getenv "PORT")]
+    (Long. s)
+    (throw (ex-info "Must set enviroment variable PORT" {}))))
+
+(defn -main
+  "Command-line main entry point."
+  []
+  (let [port   (env-port)
+        system (-> (system/system {})
+                   (update ::system/endpoint endpoint/with-port port)
+                   component/start)]
+    (log/info :msg "System started" :port port)
+    (println "Service started at port" port)
+    (endpoint/join (::system/endpoint system))))
```

Comment:
`:endpoint`, not `::system/endpoint`.

BTW, what's the benefit of `endpoint/join`?